### DISCUSSION TITLE
How to enable mosquitto password file and acl file access control?

### DISCUSSION THREAD

**Discussion by iglara** [halfgaar/FlashMQ#16] [Q&A] [answered]
Hi,
Can you please help me with explaining how to use existing mosquitto password and acl files in FlashMQ config?
I mean I got it, set it in the config file, but not working.
In one side I see error, that the code expects $6 and $ separated 3 column lines in the password file, but what I have is $7 and more than 3 columns. This is reflected in the log as well:
`Dropping invalid username/password line: 'mqttexplorer:$7$101$...'. Error: Invalid line format in 'mqttexplorer:$7$101$FI..something$0eE..something=='. Expected three fields separated by '$'`

Also I couldn't figure out what to do with this in the config file, can you pls help to explain if I need to define this for mosquitto password file to work? If so, what is the proper config here if I followed the docker build instructions?
```
# plugin supports Mosquitto version 2 or FlashMQ native.
plugin /path/to/plugin.so

```
Overall I like the solution, I am just not experienced enough to figure out this part myself. My FlashMQ is running in docker, but only with the allow_anonymous true option, since the password and acl file not working for me.

I appreciate your help.

**Comment by halfgaar** [comment, marked as answer]
Apparently Mosquitto introduced a new hashing format. I'll get on supporting that. https://github.com/halfgaar/FlashMQ/issues/17

In the mean time, you can make passwords with `mosquitto_passwd -H sha512`.

As for the plugin: you don't need that for the password file. The plugin is for when you write C(++) code to do the authentication.

If you do get into plugin development, there are many caveats when using Docker or AppImage.

**Reply by iglara** [reply]
Thanks for the quick reply. I could generate a new password file, which works now.

**Comment by charugaa** [comment]
I installed the server and it works, but I don't understand how to enable the username and password to connect to the mqtt server. Can you please provide an example of a config file?

**Reply by halfgaar** [reply]
Well, it should just work. You make a file, like `/etc/flashmq/users` containing entries created by `mosquitto_passwd`, like:

```
wiebe:$7$101$oW4vum9VREb/0aH6K3ge7kY9u26JboPDpF+9SKjd0fdz6SXidJ9n+TtBbg5n0ImfM/LkIEn6BZxT78Wn6Uzgvw==$Sc31amckhadG9Sd7/qTR8L+ROxljBLCMTrFj/Isb0rikvolsWnucdfDDY3bcIk03Nm/ymzfenE7LuTtgbz3wkA==
```

Then point to that file using [mosquitto_password_file](https://www.flashmq.org/man/flashmq.conf.5#mosquitto_password_file) in `flashmq.conf`

If that doesn't work, can you show your config?
